plugins/janus_videoroom.c

```diff
@@ -2292,6 +2269,36 @@ void janus_videoroom_incoming_rtp(janus_plugin_session *handle, int video, char
 	if(!session || session->destroyed || !session->participant || session->participant_type != janus_videoroom_p_type_publisher)
 		return;
 	janus_videoroom_participant *participant = (janus_videoroom_participant *)session->participant;
+	if(participant->audio_active) {
+		int level = 0;
+		if(janus_rtp_header_extension_parse_audio_level(buf, len, participant->audio_level_extmap_id, &level) == 0) {
+			// JANUS_LOG(LOG_INFO, "Audio level is %d\n", level);
+			participant->audio_dBov_sum = participant->audio_dBov_sum + level;
+			participant->audio_active_packets = participant->audio_active_packets + 1;
+		// 2 seconds of talking (100 packets) with average of ~25 dBow
+			if(participant->audio_active_packets == 100) {
+				if(participant->audio_dBov_sum < 2500) {
+					// Notify participants
+					json_t *event = json_object();
+					json_object_set_new(event, "event", json_string("talking"));
+					json_object_set_new(event, "user", json_string(participant->display));
+					janus_videoroom_notify_participants(participant, event);
+					json_decref(event);
+					JANUS_LOG(LOG_ERR, "AVG audio_level %f\n", (float)participant->audio_dBov_sum/(float)participant->audio_active_packets);
+					/* Also notify event handlers */
+					if(notify_events && gateway->events_is_enabled()) {
+						json_t *info = json_object();
+						json_object_set_new(info, "event", json_string("talking"));
+						json_object_set_new(info, "room", json_integer(participant->room->room_id));
+						json_object_set_new(info, "user", json_string(participant->display));
+						gateway->notify_event(&janus_videoroom_plugin, session->handle, info);
```

Comment:
I'm wondering if we do need the `session->handle` here: without it, it becomes an event the plugin generates by itself and not strictly associated with a specific user (e.g., as a room being created). On the other end, it is triggered by incoming media from a specific user, so it's probably fine to keep it too.